hare/algorithm_test.go

```diff
@@ -464,42 +470,34 @@ func TestConsensusProcess_currentRound(t *testing.T) {
 func TestConsensusProcess_onEarlyMessage(t *testing.T) {
 	r := require.New(t)
 	proc := generateConsensusProcess(t)
-	m := BuildPreRoundMsg(generateSigning(t), NewDefaultEmptySet())
+	m := BuildPreRoundMsg(generateSigning(t), NewDefaultEmptySet(), nil)
 	proc.advanceToNextRound(context.TODO())
 	proc.onEarlyMessage(context.TODO(), buildMessage(nil))
 	r.Equal(0, len(proc.pending))
 	proc.onEarlyMessage(context.TODO(), m)
 	r.Equal(1, len(proc.pending))
 	proc.onEarlyMessage(context.TODO(), m)
 	r.Equal(1, len(proc.pending))
-	m2 := BuildPreRoundMsg(generateSigning(t), NewDefaultEmptySet())
+	m2 := BuildPreRoundMsg(generateSigning(t), NewDefaultEmptySet(), nil)
 	proc.onEarlyMessage(context.TODO(), m2)
-	m3 := BuildPreRoundMsg(generateSigning(t), NewDefaultEmptySet())
+	m3 := BuildPreRoundMsg(generateSigning(t), NewDefaultEmptySet(), nil)
 	proc.onEarlyMessage(context.TODO(), m3)
 	r.Equal(3, len(proc.pending))
 	proc.onRoundBegin(context.TODO())
 
 	// make sure we wait enough for the go routine to be executed
-	timeout := time.NewTimer(1 * time.Second)
-	tk := time.NewTicker(100 * time.Microsecond)
-	select {
-	case <-timeout.C:
-		r.Equal(0, len(proc.pending))
-	case <-tk.C:
-		if 0 == len(proc.pending) {
-			return
-		}
-	}
+	time.Sleep(100 * time.Microsecond)
+	r.Zero(len(proc.pending))
```

Comment:
```suggestion
	r.Len(proc.pending, 0)
```
This will produce a nicer error message (incl. the actual length).